### PR TITLE
Feature/order endpoints with db

### PR DESCRIPTION
**Pull request by JbreenJboor1212**
Updated the **Order** endpoint to use a real database instead of a fake one, and adjusted all related endpoints to reflect this change.

### CODE REVIEW

**Review by Habeeb-Awawda — APPROVED** [review]

File: app/schemas.py [hunk 1]
```diff
@@ -273,3 +273,113 @@ class GetProductResponseModel(ProductBaseModel):
...
+            Decimal: lambda v: str(v)
+        }
+    
+
+class UpdateOrderStatusResponseModel(BaseModel):
+    id: UUID
+    user_id: UUID
+    total_price: Decimal
+    status: str
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
```
Comment (Mohammed-samaheen) [inline app/schemas.py]: create date for update model?

File: app/services/order_service.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            id=new_order.id,
+            user_id=new_order.user_id,
+            status=pending_status.name,
+            total_price=new_order.total_price,
+            created_at=new_order.created_at
+        )
+
+        return response_data
+
+    def _get_pending_status(self):
```
Comment (Mohammed-samaheen) [inline app/services/order_service.py:38]: It's not ideal to keep fetching statuses for each order. Instead, we should load all the statuses once into memory (query all statuses and store them in a dictionary with the status name as the key). Don't worry if you don't fully understand this or know how to implement it right now—you can keep it as is for this phase.

File: app/services/order_service.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+from sqlalchemy.orm import Session,joinedload
+from app.models import Order,Product,Status,OrderProduct
+from app.schemas import CreateOrderRequestModel, CreateOrderResponseModel, GetOrderResponseModel, OrderProductBaseModel, UpdateOrderStatusResponseModel
+
+
+class OrderService:
+    def __init__(self, db: Session):
+        self.db = db
+
+    def create_order(self, user_id: UUID, order_request: CreateOrderRequestModel) -> CreateOrderResponseModel:
```
Comment (Mohammed-samaheen) [inline app/services/order_service.py:14]: No need for all this complexity and validation, that is the reason why we use ORM, instead what we should do is only pass the order DB model with all its order product, and in a single line, you can create the order, please read this [example](https://docs.sqlalchemy.org/en/20/orm/extensions/orderinglist.html).
long story short you should update your order DB model to use a relationship something like
```
class Order(Base):
    __tablename__ = 'orders'
    
    # other fileds
    order_products = relationship("OrderProduct", back_populates="order")
```

and in order product model something like 
```
    order = relationship("Order", back_populates="order_products")
```

please update all other DB functions to use this approach

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: app/schemas.py (hunk 1 above)
Comment (JbreenJboor1212) [reply app/schemas.py]: عامل update

Comment (JbreenJboor1212) [reply app/schemas.py]: in line 350

File: app/services/order_service.py (hunk 2 above)
Comment (JbreenJboor1212) [reply app/services/order_service.py:38]: good

File: app/services/order_service.py (hunk 3 above)
Comment (JbreenJboor1212) [reply app/services/order_service.py:14]: بس انا عاملهم هيك ال model , و مستخدم ال orm مش فاهم وين المشكلة ؟

Comment (JbreenJboor1212) [reply app/services/order_service.py:14]: Please check if the change meets your requirements

Comment (Mohammed-samaheen) [reply app/services/order_service.py:14]: I see what you did, usually all these validations we added it to a trigger function in DB, and we only insert the order itself, anyway Don't worry if you don't fully understand this or know how to implement it right now.

**Review by Mohammed-samaheen — DISMISSED** [review]

File: app/services/order_service.py (hunk 3 above)
Comment (JbreenJboor1212) [reply app/services/order_service.py:14]: I did change after from this changes at same reviews are u see ? or return to prev changes ?

Comment (Mohammed-samaheen) [reply app/services/order_service.py:14]: You can revert the previous changes, but you should also update the product stock after creating the order.

**Comment by JbreenJboor1212** [comment]
ما انا عملت التحديث , عدلته و عملت زي ما طلبت .

File: app/services/order_service.py (hunk 3 above)
Comment (JbreenJboor1212) [reply app/services/order_service.py:14]: المخزون شغال صح و جربته الان ولكن انا ارجع ل قبل ما احدث ولا اوخذ التحديث الي عملته انا ؟

Comment (JbreenJboor1212) [reply app/services/order_service.py:14]: I Did That>

**Review by Mohammed-samaheen — APPROVED** [review]

**Pull request merged**